src/qt/transactionrecord.cpp

```diff
@@ -53,30 +53,56 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
     }
 
     if (wtx.IsZerocoinSpend() || isAllSigmaSpendFromMe) {
+        CAmount nTxFee = nDebit - wtx.GetValueOut();
+        bool first = true;
+
+        bool isAllToMe = true;
         for (const CTxOut& txout : wtx.vout) {
-            if (wtx.IsChange(txout)) {
-                continue;
+            if (!wallet->IsMine(txout)) {
+                isAllToMe = false;
+                break;
             }
-            isminetype mine = wallet->IsMine(txout);
+        }
 
+        if(isAllToMe){
             TransactionRecord sub(hash, nTime);
             CTxDestination address;
+            sub.involvesWatchAddress = ISMINE_WATCH_ONLY;
+            if (ExtractDestination(wtx.vout[0].scriptPubKey, address) && IsMine(*wallet, address)) {
+                sub.type = TransactionRecord::SpendToSelf;
+                sub.address = CBitcoinAddress(address).ToString();
+                sub.credit = -nTxFee;
```

Comment:
I think this is not right. It should be `sub.debit = -nTxFee;` instead.

Reply:
Looks you are correct.

Reply:
sub.credit = -nTxFee; this way in transaction list it is shown red, I think better to keep this;